lib/Doctrine/DBAL/Query/Expression/CompositeExpression.php

```diff
@@ -36,6 +37,8 @@ class CompositeExpression implements Countable
     private $parts = [];
 
     /**
+     * @deprecated Use the and() / or() factory methods.
```

Comment:
We shouldn't deprecate a method that is not going to be removed. It's better to mark it `@internal` instead.

Reply:
Done in ff626cab4ae5e0d134a0571bf816f3f0269942bb.